crates/supervisor/core/src/logindexer/indexer.rs

```diff
@@ -0,0 +1,175 @@
+    /// Component that persists parsed log entries to storage.
+    pub log_writer: Arc<dyn LogStorageWriter>,
+}
+
+impl LogIndexer {
+    /// Creates a new [`LogIndexer`] with the given receipt provider and state manager.
+    ///
+    /// # Arguments
+    /// - `receipt_provider`: Shared reference to a component capable of fetching receipts.
+    /// - `log_writer`: Shared reference to the storage layer for persisting parsed logs.
+    pub fn new(
+        receipt_provider: Arc<dyn ReceiptProvider>,
+        log_writer: Arc<dyn LogStorageWriter>,
+    ) -> Self {
+        Self { receipt_provider, log_writer }
+    }
+
+    /// Processes the logs of a given block and stores them into the state manager.
+    ///
+    /// This function:
+    /// - Fetches all receipts for the given block from the specified chain.
+    /// - Iterates through all logs in all receipts.
+    /// - For each log, computes a hash from the log and optionally parses an [`ExecutingMessage`].
+    /// - Records each [`Log`] including the message if found.
+    /// - Saves all log entries atomically using the [`LogStorageWriter`].
+    ///
+    /// # Arguments
+    /// - `block`: Metadata about the block being processed.
+    pub async fn process_and_store_logs(&self, block: &BlockInfo) -> Result<(), LogIndexerError> {
+        let receipts = self.receipt_provider.fetch_receipts(block.hash).await?;
+
+        let mut log_entries = Vec::new();
+        let mut log_index: u32 = 0;
+
+        for receipt in receipts {
+            for log in receipt.logs() {
+                let log_hash = log_to_log_hash(log);
+
+                let executing_message = parse_log_to_executing_message(log).map(|msg| {
+                    let payload_hash =
+                        payload_hash_to_log_hash(msg.payloadHash, msg.identifier.origin);
+                    ExecutingMessage {
+                        chain_id: msg.identifier.chainId.try_into().unwrap(),
+                        block_number: msg.identifier.blockNumber.try_into().unwrap(),
+                        log_index: msg.identifier.logIndex.try_into().unwrap(),
+                        timestamp: msg.identifier.timestamp.try_into().unwrap(),
+                        hash: payload_hash,
+                    }
+                });
+
+                log_entries.push(Log { index: log_index, hash: log_hash, executing_message });
+
+                log_index += 1;
+            }
+        }
+
+        self.log_writer.store_block_logs(block, log_entries)?;
+
+        Ok(())
+    }
+}
+
+/// Error type for the [`LogIndexer`].
+#[derive(Error, Debug)]
+pub enum LogIndexerError {
+    /// Failed to write processed logs for a block to the state manager.
+    #[error(transparent)]
+    StateWrite(#[from] kona_supervisor_storage::StorageError),
+
+    /// Failed to fetch logs for a block from the state manager.   
+    #[error(transparent)]
+    FetchFailed(#[from] jsonrpsee::core::ClientError),
```

Comment:
```suggestion
    FetchFailed(#[from] jsonrpsee::core::ClientError),
```
same here, import in import prelude